### DISCUSSION TITLE
ios / cross platform coming?

### DISCUSSION THREAD

**Discussion by cyclingwithelephants** [advplyr/audiobookshelf-app#60] [General]
Hi, 

I was hoping to ask about the intentions of this project, specifically whether or not an iOS app might be in the works. I appreciate this is early days and want to thank you for the great work - I'm just trying to evaluate my options

**Comment by advplyr** [comment]
Yes, the iOS app is started. This repo also has the iOS code.
No eta yet, the Android app still has several critical bugs to fix. I will post updates here since this is the first discussion on ios.

**Reply by SauerDeschain** [reply]
The app for audiobooks works very well. All of my collection work well. 

The main issue is that podcasts load up as existing, but clicking then does not start playback in any situation. I've tried all of the variations, and it's almost like the play button isn't attached to the call for it to start. 

Keep up the great work and thank you!

**Reply by IamNotASmartMan** [reply]
I had the same issue but since the last update it’s playing fine

**Comment by advplyr** [comment]
I've got the app running on iOS with everything excluding downloads/offline play and sleep timer. Also, I wasn't able to get the now playing info working yet to control the audio from the lock screen, but the audiobook does play in the background.

My apple developer account is pending approval now. Once it is approved I will release this version as an early release if apple supports that.

**Reply by advplyr** [reply]
On the second rejection they said it appears the app is not ready for production, I'm guessing because I labeled the version 0.9.35-beta. Apple app store doesn't have open beta testing like android, so I'm hesitant to release this early and get a lot of comments and reviews about features missing.
It may be a while to finish all those features unfortunately.

**Reply by 7aklhz** [reply]
Hi. I just started using your project. Great work !! Can you tell me if offline mode is planned for ios ? This is really a must have feature when I'm using the app on plane/subway, etc.

**Reply by advplyr** [reply]
Offline mode is planned yes. So far I have gotten the iOS app to download an audiobook to the device but nothing farther then that.
We are looking for some developers to help out with the iOS app to speed up the process.

**Reply by 7aklhz** [reply]
Ok, good to hear. Thanks and looking forward to that feature.

**Reply by vPeteWalker** [reply]
I'd love to see a download/offline mode. Thank you.

**Reply by DrBazUK** [reply]
Thanks for the development effort. This is great but would echo earlier comments that downloading option for iOS would be the icing on the cake.

**Reply by c-hri-s** [reply]
@DrBazUK the iOS app supports downloading, it's been in there a while

**Reply by DrBazUK** [reply]
> @DrBazUK the iOS app supports downloading, it's been in there a while

Thanks, I think I must have either been doing something wrong or managed to grab a version of the app where downloading was not working. I removed the app and reinstalled from Testflight and happy to confirm that I can download locally to the phone. It seems I wasn't the only one from the issues log so wonder if there was a version where downloading was briefly borked.

**Reply by Jmanko16** [reply]
Downloading is broken again in most recent update

**Reply by helio4k** [reply]
Yes, I noticed this morning that the download is broken. Also on occasion it will crash the App it self.

**Reply by Jmanko16** [reply]
What's weird is with the new server update, when I click download now nothing happens but after a few minutes I get a download complete message.  The download spinner bar with progress never shows up.  Very weird.

**Reply by helio4k** [reply]
Yep, just updated the server and the ios app is not crashing anymore but there is no progress of the download, it simply says downloaded :) 
a weird bug but I will take it over not being able to download books to my phone.

**Reply by benonymity** [reply]
Yep, should be a simple fix. Watch #537 for resolution, should be working in the next update.

**Reply by mfgWHOAMI** [reply]
Everthing works and looks very fine. I only missed the Airplay function. 
Do you have en donate function for your work?

**Comment by advplyr** [comment]
Just to open a new post, I explained why I'm not putting the ios on the app store yet here: https://github.com/advplyr/audiobookshelf-app/discussions/60#discussioncomment-1925438

Update: the app got approved for beta testing and can be installed through this link: https://testflight.apple.com/join/wiic7QIW

**Reply by cyclingwithelephants** [reply]
I'd be more than happy to do this

**Reply by pittbull** [reply]
Getting "Safari cannot open the page because the address is invalid" when clicking on the testflight-link.

**Comment by b3nd1k** [comment]
Hey! I downloaded the iOS-version a few hours ago and I like, but there is definitely room for improvement. Can't wait to see how it improves over time. I ran into a few bugs that mostly a force close of the app seemed to fix. I do have a few suggestions I hope you take into consideration though.

- The first is if I am currently playing one book and click "Play stream" on another book, the book I am currently listening to should be paused until the other one starts. As of now, the current book will keep playing until the new one starts. 
- Where I've the rewind, play and skip forward buttont here should also be an "X" or something to close the book. 
- When it comes to the design, the iPhone app needs some adjustments. At the bottom left and the bottom right where it shows the progress of the audio book and the total time, some of the digits are not visible because of the curves of the display. 
- In the app there is two sections "Recently added" and "Recent series" and below a big grey field. For books that are part of collection where for example the first book in the series have been listened to, I could see you ad a "Up next" section here. Otherwise The play/progress bar could probably be a little bit bigger.

**Reply by mxdalchau** [reply]
when we get a update? a nice app on android ^^but on ios on iphone 13 it is useless

**Reply by advplyr** [reply]
What do you mean useless? That it is not working at all, or that it is missing too many features still?

**Reply by mxdalchau** [reply]
i have a iphone 13 pro and i can stream ... a title but i can´t see the player.. with play button an the other buttons... no sleep timer :(

**Reply by advplyr** [reply]
You can't see the player? You mean you can't see the player on the lockscreen?

**Reply by mxdalchau** [reply]
i can´t see the player in the programm... the audiobook begin to stream i heal the voice but cant see the player in the app

**Reply by advplyr** [reply]
That's odd.. If you can share a screenshot in a new thread of this discussion that would be appreciated

**Reply by mxdalchau** [reply]
![IMG_62BA994B30C0-1](https://user-images.githubusercontent.com/28609519/152697916-17e21f46-cdc8-4952-8516-0a60730211d1.jpeg)

**Reply by advplyr** [reply]
Ah I see. I think I fixed this in the android app. I will try to get a new release out for this asap.

**Reply by mxdalchau** [reply]
thanks.. in android it is beautiful.. but on ios is useless :)

i would pay for it.. the best server app concept^^

**Reply by advplyr** [reply]
The new version should be available on test flight to update to. v0.9.40

**Reply by mxdalchau** [reply]
if iwant  to resume the book on my iphone 13 pro... ... i press play button... it starts to load.. thats all :(. (m4b 8h file)

**Comment by porternw** [comment]
Hi. I just loaded the beta version today (IOS 15.3.1). I am able to connect to my server from the ios app without trouble, and it loads my library. I can see all of the controls. When I choose to stream a title it doesn't play at all. The spinner goes around for a few seconds and stops. The titles play fine when I play from a browser. Thanks for working on this, I really like the server.

**Reply by porternw** [reply]
To clarify, it plays from a web browser on a computer. IOS safari doesn't work either.

**Reply by advplyr** [reply]
Hey what is the version of the server you are using?

**Comment by Kussie** [comment]
Been using the app on iOS the last few nights and when as the person above reported pressing the play button doesn’t always do anything, I sometimes need to press 5 or more times before it starts playing.

I have also found the sleep timer does not work for me, I set it for 30 minutes and I will fall asleep and wake up a few hours later to it still playing.


App version: 0.9.40
Server: 1.7.2 (Docker)

**Reply by rasmuslos** [reply]
The sleep timer will be working in the next version. The native player will receive some updates, most importantly the way the playback is started will change. Hopefully, this will fix the issue

**Comment by mxdalchau** [comment]
nice :)  is there a timeline, when the next version is coming?

**Comment by jparrish88** [comment]
Hi All, 

I just got the ios app up and running. Are downloads in the iOS app working? If so, can they be accessed with a user account? I've switched the account from a user to an administrator, but didn't see any changes. The account has the ability to download, as far as the website settings say. 

If everything is suppose to work, do I need to do anything to enable downloads in iOS? Is there a permissions setting of some kind? I just don't see the same downloads button in ios as I do in Android. 

Thanks in advance for anyhelp.

**Reply by advplyr** [reply]
Downloads haven't been implemented yet for iOS

**Reply by fistwho** [reply]
Any ETA for this Feature ? Thank you for this amazing Project 🙏

**Reply by STX2k** [reply]
+1 for this feature soon™

**Comment by renocz** [comment]
Hi guys, thanks for all the good work!
I have an issue with the IOS app (previous and current version). I have a reverse proxy (NPM) setup and it works well on a navigator (websocket support activated). However when i use the address as Server address in the ios app i have an error message "Failed to ping server". If i'm on my local network and I use the local address it works. Any idea ? A bug? Thanks a lot.
![image](https://user-images.githubusercontent.com/22596767/164242321-89aa01e3-c536-4c14-8ae2-f586bc85bcb7.png)

**Reply by benonymity** [reply]
Have you configured the local DNS for your network to redirect your domain to the server you're using? Though counterintuitive, you can't access a domain in the network you're serving it from without further configuration. Can you access the web interface from a computer on the local network? (i.e. Is the app the only issue?) If port forwarded, do you get the same error when on another network/cellular? Glad you're enjoying the app, it's really awesome!

**Reply by renocz** [reply]
Hi Ben,
Thanks for your reply. 
Actually I can access all my apps using the domain on my lan but just to confirm your point, it doesn't work as well using cellular network. I have still the same issue.
However, I can access to the app using my domain if I go through a web browser that's why I suspect an issue with the mobile app

**Reply by benonymity** [reply]
Hey,
Got it. And I assume if you can access it locally you're running the latest server and app versions released last night? I can confirm the issue, I just logged in and out multiple times over cellular and once got the same "failed to ping" response. I checked the code and the current ping timeout seems to be 1 second, and that delay getting hit over cellular is probably the issue. I can submit a PR to increase the timeout. I've noticed similar issues with logging in, occasionally I have to do it more than once even though the credentials are correct, probably the same root cause. Thanks for bringing it up!

**Reply by renocz** [reply]
My pleasure ☺️
Thanks for your reactivity !

**Reply by vPeteWalker** [reply]
Hi there. I'm having the same exact issue on IOS using 0.9.51-beta. I can't connect to the server internally or externally via FQDN or IP address. I rolled back to 0.9.50-beta and I was able to connect over IP internally, but not FQDN. Both IP and FQDN both work with the browser, both internally and externally. If I add the server in using IP in 0.9.50, and then upgrade to 0.9.51, it works. Thank you.

**Reply by PlzSpamMeNot** [reply]
Hi, I have the same problem but with the android app. I tried it on several devices, locally and externally.
I tried over IP and DNS name. Nothing work for me.

Over web I could access the server from every where and from every device.

The server is new and also the App was installed the first time on the devices, so no experience with the older versions.
I use the App Version 0.9.57-beta over Play Store (from 27.08.2022)

**Comment by mbentham** [comment]
Hey, really liking your work. I’m running v0.9.41 of the iOS app with v2.0.0 server through Docker and I’ve found that my progress or part of my progress in an audiobook seems to be lost when switching from the app to the web player.

My playback speed setting is also reset to 1x every time I exit and reopen the app, and I’ve noticed the sound quality when playing back at higher speeds isn’t great, sounding distorted and tinny.

Overall it’s great work and looking very promising

**Reply by advplyr** [reply]
I believe some of these you listed are fixed in the latest updates of server and apps, but let me know or open up an issue for these if they still persist.

**Comment by pittbull** [comment]
I am trying to download the app via the Testflight link but I'm getting "address not valid". Is there anything else I need to do?

**Reply by B0rax** [reply]
https://testflight.apple.com/join/wiic7QIW

This link should work.

**Reply by pittbull** [reply]
Sorry for not being clear: That page loads, but when I click "Join the beta" it responds "Safari cannot open the page because the address is invalid."

**Reply by B0rax** [reply]
Did you try that on an iOS device? Do you have TestFlight (from the AppStore) installed?

**Reply by pittbull** [reply]
Well, that was embarrassing: recently switched phones and it hadn't reinstalled testflight. Confirming all good now.

**Comment by IamNotASmartMan** [comment]
I have the most recent TestFlight app and when I try to play a podcast, it does not play on my app and it doesn't play and the logs read
INFO
[PlaybackSessionManager] startSession: Closing open session "Here's How Politicians Pick Their Voters" for user "nathan"
Am I doing something wrong?
Audiobooks play flawlessly.

**Reply by advplyr** [reply]
Are you able to play any podcast episodes?

**Reply by IamNotASmartMan** [reply]
They play on the browser version but not in iOS

**Comment by helio4k** [comment]
Dame I am so happy and impressed about this application docker, android and apple top notch. 
thank you all for hard work that was put in for the community.

**Comment by PoepieFloepie** [comment]
On the ipad pro i get no more then 20 covers the rest is transparent and if i press one then it's going to the home screen
in list mode i see every cover
please fix....

**Reply by advplyr** [reply]
This is fixed in the latest release 0.9.50

**Comment by PoepieFloepie** [comment]
i have a other problem with ipad pro ..
cant play audiobook ,i press green button nothing happens (latest beta release 0.9.47)(but on the same beta it works on the iphone)
i downgrade to version 0.9.42 and everything works again..
wierd...

**Reply by advplyr** [reply]
Can you try upgrading to the latest release?

**Reply by PoepieFloepie** [reply]
Working...big thank..

**Comment by akamel001** [comment]
Hi! Any plans to integrate the app with Apple CarPlay? Really love this app and want to make a full switch to it, but the only hurdle is CarPlay. Would love to see that integrated!

**Reply by Manantra** [reply]
Same for me.. I would love to switch completely, but I'm missing CarPlay Support.

**Reply by philhagger** [reply]
This is the one feature that stops me daily driving over Audible.

**Reply by norjms** [reply]
Another vote for CarPlay everything else is working great.

**Reply by michaelkrieger** [reply]
More carplay votes.  Carplay is on most new cars and makes this a daily driver.

**Comment by anthr76** [comment]
Are there any plans to implement Chromecast on the iOS app?

**Comment by MrDaft** [comment]
Hi,
thank you for your great work!
but can I ask about placing timeline higher and making it a little bit narrow? It nearly imposible in the begining of the book grab a timeline "ball" it is always trying to swipe me back

**Reply by MadWalnut** [reply]
Yes, same for me.

**Reply by helio4k** [reply]
Same for me, also I have notice that that during listening to my audiobooks local or streaming if some one calls and after that I go back to the book it throws me back sometimes 30 minutes back.  Super annoying.

**Comment by HuLuXi** [comment]
i found that if the audio is .wma, it cant be play with ios app 
but if i use pc explorer and safari ipad, it play normally.
and i play .m4a it worked dine 
is that ios app not adaptable for .wma

my docker is v2.2.6

**Reply by borislavmilijic** [reply]
I have the same problem with my iPad Mini - but in my case the files are .opus
My android devices play them, via Browser also, but the iOS App seems to have a problem with them. Logs also don't say a lot in this case. @advplyr

**Comment by srvoets** [comment]
Been using this for a week or two. Unraid, traefik.

iOS app is pretty good but needs a few things thought I would just make a list 

- CarPlay support

- mark podcast as "done" from player menu

**Reply by meichthys** [reply]
Ooo. Carplay support would be awesome!

**Comment by taxilian** [comment]
I've got this set up and have been using it for a week -- fantastic! It's not perfect yet, but it's impressively close to what I want.  I don't care about the sleep timer, but here are the things I've seen in the "can be improved" category:

* I have had it lose my place after pausing a few times -- it usually works, but sometimes it will jump back a bit; I actually like it jumping back a short bit, but once or twice it went back well over a chapter (~25 minutes in this book) which was really a pain to get back
* While trying to get back to my place I found that scrubbing doesn't work real well, mainly because it shows the progress in the book instead of progress in the current chapter -- that's really annoying and differs from the other audiobook apps I've used which always show the current chapter. It's just impossible to scrub with reasonable accuracy at that scale.
* The views aren't quite intuitive for me; it's hard to find what I want. I'm not sure how to improve that so I'll keep brainstorming and maybe pitch in with a MR when I have better ideas
* I prefer to download books and then play them locally like with audible -- this really doesn't seem optimized for that, which seems strange. In particular it takes a lot of taps/clicks to delete a book that you downloaded.

All that said, I just barely found this project and I am super stoked about it -- it's in a language that I can work in so I hope to contribute and it is *exactly* what I've been missing for quite some time. I have calibre for ebooks but haven't had a good option for my audiobooks.  Keep up the great work!

**Reply by advplyr** [reply]
Your help would be great! I don't have much experience with iOS so have been relying a lot on contributions there.
As for the chapter track part you can toggle on the chapter track view in the menu at the top right of the fullscreen player.

We can also use some help in the design. For many features that is the most difficult to figure out. 

The discord & matrix servers are active and are where a lot of discussion around these things happen.

**Reply by helio4k** [reply]
I also experienced some issues that have been described by OP, I think that the most annoying thing is that when I am listening to the audiobook and someone calls me then for whatever reason the audiobook will rewind anywhere from 25 - 40 min little bit annoying. but overall great application.
Keep up the awesome work.

**Reply by taxilian** [reply]
Very interesting -- Why is "chapter track" not the default, like it is on every other audiobook player I've used? =]

I'm on the discord and happy to help as I can; my time is limited, but I do have some experience with UX, though I wouldn't call myself an expert, and I use audiobooks *extensively* so I can at least give feedback on what would feel natural.

Feel free to ping me on discord any time for anything I can help with.

**Comment by yzwko** [comment]
Just downloaded this, but I can't seem to find the option to open RSS feeds from the app. Any plans to add this in the future?

**Reply by advplyr** [reply]
Sure, we can do that.

**Comment by Jmanko16** [comment]
Really liking the app.  Not sure if this or discord is the right place for feature requests but I would like to see an option to download file on the app (the web login has this, but not app).  I'm using audiobookshelf to catalogue all my ebooks in pdf and epub, and much easier to have a download button and read in another app until the readers are improved.  For less technical (Aka family) members I do not think navigating the web app and downloading files would be an option.  An implementation like Calibre-web where you click download and gives option (.m4b, .mp3, .epub, .pdf) to save locally on iOS would be amazing.

Keep up the great work!

**Comment by Jdiesel87** [comment]
Not asking for an ETA but I am interested in learning about the barriers to getting the app on the Apple App Store. Is the approval process more stringent? Is it difficult to distance the app from piracy uses, etc? Or is it that the app just not at a state the would warrant a submission through the approval process?

I am not an Apple user myself but most of my family is. I've been hesitant to have them go the Testflight route as I don't want to do the IT for them.

**Reply by advplyr** [reply]
I think we're really close at this point. There is nothing about the App store approval that is holding it up, I just didn't want to publish an app that wasn't solid and start getting negative reviews because we put it up too early.

The last major thing I want to have done before this goes up is Apple Car play.
I don't use the iOS app outside of regular testing and dev so I'll have to look through the bug list to see which bugs need fixing before production release. There is one bug with iOS with the seek position jumping back that should be resolved before production.

It is probably a good idea to get a checklist going since we are so close now but I'm not very organized in that regard.

**Reply by Jmanko16** [reply]
CarPlay support would be awesome (consider making the "bookmark" button available via the CarPlay interface.

only other feature I would love is ability to download epub/pdf from app with a button to use external readers.

The app is great overall.  Glad to hear the going backwards problem will be fixed as well.  I get it occasionally but can't reproduce it.

**Reply by maodelg** [reply]
Thanks for all your hard work. Out of all my self-hosted services, this is by far the BEST! Crossing my fingers for the full iOS release soon!

**Comment by norjms** [comment]
My biggest issue is where series don’t display on an iPhone once you have
more than a few. I did notice that it displays fine on an iPad however.

**Comment by c-hri-s** [comment]
Can I just echo the need for carplay support, this would be an amazing addition.

As someone who likes to listen to books in multiple places, I'm hoping the app is what can help synchronise my listening across multple platforms (specifically web and iOS). Is there any detail on how/when it syncs position data? A brief test suggested that it does ... but it wasn't clear to me how/when, as sometimes it didn't seem to notice I'd played some on the phone/web.

My use case is an internal server at home (not exposed to the web), So would plan on downloading books to the iOS app, listening while out-and-about/driving, and then my position being synched to the server when I was at home so that I could continue the book up on the web interface.

**Comment by andrewdunndev** [comment]
Is it possible to install this on an AppleTV (tvOS)? I noticed on the tvOS test flight app I can't seem to type all the characters of the invite, which I suspect is `wiic7QIW`.

Edit: Apologies, I see it's flagged for mobile devices and not tvOS. I'd say it'd be _nice_ to be able to use it on tvOS but it should not be a priority compared to doing a proper release (as [talked about here](https://github.com/advplyr/audiobookshelf-app/discussions/60#discussioncomment-4613301))

**Comment by Quildar** [comment]
First off, I'm enjoying your app, and I absolutely love your backend and web portal. Things integrate really well with all my tagged M4B files. I would like to offer a couple of feature requests and one that may be a bug.

Requests: 
Make the icon at the top link to a page that shows currently downloading books along with details like the amount of data downloaded and estimation to completion. Also, the ability to cancel a download from the same page.
Make the books swipeable to get to options. For example, swipe right on a book to download/delete download, and swipe left to make as read/un-read.

Possibly a bug:
When I tell a file to download the percentage hangs at zero then jumps to 50% or just finishes.

Either way thanks for all your hard work!!!

Oh and +1 for carplay. Love me some carplay.

**Comment by helio4k** [comment]
I also am experiencing the download not working and on some occasions will crash the Audiobookshelf application (force close)

**Reply by astrodad** [reply]
I was having issues with this today. The app was showing 0% download, on the book screen, and in the Downloads screen. I left the app open for about 20 minutes anyway. The book screen showed "Stream". I force-closed the app and then the book screen showed "Play". And I was able to go offline and play the book.

**Comment by Insan0** [comment]
I just got this yesterday.  When at home, i can connect to the server with the iOS app by connecting to server address 172.160.0.3:13378.  When I'm out of the house, I use Tailscale to connect to the server (address is `[machine].[uniqueTS#].ts.net:13378`).  The Tailscale address works fine when I access the web app through Safari, but not when I use the iOS app.  I get an error "Failed to ping server".    

Same error when I'm at home and use the Tailscale address, even through my local DNS points to the local address.  Again, the Tailscale address works at home when using the web app through Safari.  Hope that makes sense.

**Reply by advplyr** [reply]
We upgraded the deployment target to iOS 14 for this last update and it seems like iOS is now requiring `https`. Everyone using `http` for external connections are no longer able to connect.

**Reply by Insan0** [reply]
OK, so I'm just getting into self hosting and pretty new to all of this.  So my background is: this all seems like a miracle :)  

My understanding with Tailscale is that it's supposed to act like it's on a local network when remote.  Maybe to clarify with a matrix: 
                                                                                                 
Using: `http://[machine].[uniqueTS#].ts.net:13378`  
- At Home
  - App: Error (not expected)
  - Safari: Good
- Remote
  - App: Error (not expected)
  - Safari: Good

Using: `http://172.16.0.3:13378` 
- At Home
  - App: Good
  - Safari: Good
- Remote
  - App: Error (expected)
  - Safari: Error (expected)

Not trying to challenge you, just trying to clarify incase my first message wasn't clear.   If it is indeed an issue with `http` vs `https`, is there a configuration I need to enable?  I didn't see anything in the server settings. I've installed through Docker.  Thanks again.

**Reply by Insan0** [reply]
One last thing, this is really amazing and great work!!

**Reply by advplyr** [reply]
I'm not familiar with tailscale but there are a bunch of people in the Discord that use it. Since you didn't have this setup in the previous version it's hard for me to say if this is hitting the http issue that others have been reporting.
You're most likely to get assistance in Discord or Matrix

**Reply by Harwood** [reply]
With Tailscale I’ve noticed the same. However using just the hostname like `http://[machine]:13378` works just fine both local and remote. Just be sure to use the hostname configured on https://login.tailscale.com/admin/machines

Another option would be to use the `100.x.y.z` address which is statically defined within the Tailscale network. https://tailscale.com/kb/1015/100.x-addresses/

**Comment by c-hri-s** [comment]
If it helps I have 0.9.61 and I'm connected to my local (LAN) http://-based instance?

**Comment by Batnun** [comment]
Is a new iOS beta version coming soon? The current testflight build will expire in 19 days...

**Reply by advplyr** [reply]
Yes

**Comment by tientcheufortune** [comment]
Hello everone, how can i get testfight invitation code to install [audiobookshelf] apps ?
my email: christiantchindebe@outlook.fr

**Reply by advplyr** [reply]
You just have to follow the link on the website on your device. Here https://testflight.apple.com/join/wiic7QIW

**Comment by mxsrm** [comment]
Are you planing to add opus codec support for iOS? E.g. using https://github.com/alta/swift-opus. I have my whole library neatly encoded in opus to save space while retaining great quality and currently everything gets transcoded on iOS which just does not work good.

**Reply by advplyr** [reply]
I wasn't aware of that project. Thanks

**Comment by Volrody** [comment]
Hello, when I go to the link, I arrive on TestFlight, but it tells me that the beta is full. Can you add invitations? Thank you so much !

**Reply by advplyr** [reply]
I purged some old beta testers so you should be good now to join TestFlight but let me know if you still have problems.

**Comment by ScuttleSE** [comment]
Looks like the beta is full again

**Reply by advplyr** [reply]
I just purged about 300 users

**Comment by scgf** [comment]
I need volume boost. It's the only thing keeping me from using Audiobookshelf as my daily audiobook app. I have to use MP3 Player on iOS and Smart Audioplayer on Android - both of which have volume boost built in. Everything else about Audiobookshelf is perfect. Is there any chance of volume boost being added?

**Comment by troelsdk** [comment]
A few versions ago I could find an epub in my library and download it to iBooks. But that is not possible anymore? To just download the file?

**Comment by elliotmarks06** [comment]
It's saying "this beta is full" again. Is there any way to increase the user limit?

**Reply by nichwall** [reply]
No, it's a hard limit set by Apple

**Reply by nichwall** [reply]
Some beta spots were cleared this morning

**Comment by nodecentral** [comment]
Hi @advplyr , if more purges are planned I’m keen to be a tester..  thanks..

(will it ever be released via the App Store?)

**Comment by DeanHamiltonnz** [comment]
Yes I just setup the server and went to download the TestFlight  app and beta full !
wish I looked before setting up the server what a waste of time

**Reply by B0rax** [reply]
You can still use the webpage on mobile until the beta is available again in a few days.

**Reply by DeanHamiltonnz** [reply]
yes thanks but web not as easy as app for blind person  but appreciate knowing that for others

**Reply by nichwall** [reply]
Spots were cleared in the beta this morning if you're still trying to get in

**Comment by moffen666** [comment]
Any aprox on when the Beta is opening for new/more testers?

**Reply by nichwall** [reply]
Not sure. There are currently issues on Apple's end that is preventing clearing out old users

**Reply by moffen666** [reply]
Ok.

**Reply by nichwall** [reply]
New spots were opened for the beta this morning

**Comment by DeanHamiltonnz** [comment]
I’m wondering if maybe apple is deliberately holding back as it  will take business from apple  books they don’t like competition just a maybe ,

**Comment by DeanHamiltonnz** [comment]
Ps is working great on iOS web but hanging for the app for the many times stuck in hospital without signal

**Comment by nodecentral** [comment]
Yes I’m in !! - Just tried the test flight link and have been allowed to install it :-) !!!

**Comment by minskicat** [comment]
Why is this still in beta, its way more stable than many commercial apps I have used on ios recently.

**Comment by sly-fox-git** [comment]
Major problems with iOS App usability:

- After a long pause of a few minutes and when the screen is locked, the app does not resume playing. You need to navigate full route to find the file that you were playing and manually resume.
- Resume issue becomes even more of a problem while driving.

**Reply by Batnun** [reply]
Same issue here.

**Reply by sly-fox-git** [reply]
Opened a bug: https://github.com/advplyr/audiobookshelf-app/issues/844

**Comment by jerry00020001** [comment]
App expires in 3 weeks on TestFlight, any chance this will get updated by then? Also, where do we go to donate?

**Reply by advplyr** [reply]
Yep, I won't let it expire. I haven't set anything up for donations yet, thanks!

**Comment by DeanHamiltonnz** [comment]
I am using the app on iOS 17 beta and it’s working flawless for me and has been for amore than a month  Thankyou so much . I also noticed the TestFlight was getting close to expiring so glad to hear it’s not. I’m blind and have  read from my server  7 books and downloaded and read another 12 books no problems at all , it’s great Thankyou so much there’s nothing else like it 👍👍👍

**Comment by ideal2545** [comment]
Do you think you might be able to clear out a few spots on the testflight? looks like its full again

**Comment by plgrimsley** [comment]
I would like a beta slot too if possible.  If there is any chance.

**Comment by kdtt87** [comment]
Hoping for a beta spot as well.
I am an avid user on android. Would love to give it a spin on IOS too.

**Comment by nichwall** [comment]
A few thousand users who did not update the app have been removed, so there are now beta spots available (September 22)

If you have an Apple developer account and would like to sideload the IPA yourself to help keep more spots available, advplyr included the IPA in the release notes.

**Reply by jacoknapp** [reply]
Also can work if you use AltStore/altserver to sideload apps, just have to make sure you resync every 7 days, and can only have three apps active with a non-developer account

**Reply by meeDamian** [reply]
I'm one of these users. And now I have a nice library sitting there with no ability to access it on my phone - any way to regain access? 😬 I even have the app installed, but it won't run, because it expired :(.

**Reply by lukethehawk** [reply]
you can use the webapp, it works very well

**Reply by meeDamian** [reply]
> you can use the webapp, it works very well

I've followed your suggestion, and tried it, but I've discovered a few really annoying quirks:

1) sometimes playback randomly stops together with an end of a chapter/part/file and only resumes after I switch back to the website.
2) earphone controls work for "stop/pause", but if the pause takes any longer than a short moment then "resume" doesn't work, and just starts playback from some other random app.

Is there really no way to update my Audiobookshelf app? 😭

I have a rather non-insignificant library that I really want to use, but I'm, quite annoyingly, limited by inability to update my app 😢.
![image](https://github.com/advplyr/audiobookshelf-app/assets/526657/4e8a5332-58ee-458c-a022-b3e0d659e367)

**Reply by c-hri-s** [reply]
You could always downlaod the ipa file and sideload it to your phone. It'll work for a week before you need to do it again.
https://sideloadly.io/ can help you sideload it

**Reply by meeDamian** [reply]
@c-hri-s Perfect, thank you so much 😁!

**Comment by lukethehawk** [comment]
After the upgrade to iOS 17 I experienced a problem, when I push the play button the stream wont start, it remain stuck.
I'm connected, I tried with wifi and 4G but it wont start the stream.
No prob with the web app through safari
I tried to reinstall completely the app,

app v. 0.9.66
srv. v 2.4.3 (docker)

**Comment by mxsrm** [comment]
I have been using the iOS app for some weeks now (instead of Prologue) and feel it is quite stable nowadays. Maybe consider an App Store release?

**Comment by LeLawnGames** [comment]
Would love to jump in on the beta if there's still spots!

**Comment by realAbitbol** [comment]
Any way I could get into the beta ? It says it's full

**Comment by nichwall** [comment]
The beta is currently full. New information here (as of November 2023) https://www.audiobookshelf.org/faq#when-will-the-apps-be-out-of-beta
There should be some spots cleared soon since there was a new version released yesterday, but not sure how many people will not have updated when spots are cleared (probably in a few days or a week).

**Comment by jacoknapp** [comment]
Or use AltStore to sideload it and as long as you have your computer running once a week, you can renew itOn Dec 9, 2023, at 4:44 PM, c-hri-s ***@***.***> wrote:﻿
You could always downlaod the ipa file and sideload it to your phone. It'll work for a week before you need to do it again.
https://sideloadly.io/ can help you sideload it

**Comment by Rukongai** [comment]
Wondering if there is a roadmap for app moving from beta to release?

**Reply by benonymity** [reply]
Yep! https://github.com/advplyr/audiobookshelf-app/issues/541
